lib/server/auth.js

```diff
@@ -86,60 +89,72 @@ class DataAccess extends EventEmitter {
-   * @param username - username as a string
+   * @param identifier - username or token
    * @param accessGroups - an array of access groups to authorise
    *                       the user against
    */
-  authorise(username, accessGroups) {
-    if (!username) {
-      throw new ReferenceError('Username is required');
+  authorise(identifier, accessGroups) {
+    if (!identifier) {
+      throw new ReferenceError('Identifier is required');
     }
-    let original_username = username;
-    username = this._getName(username);
-    if (!username) {
-      this.emit(AUTH_FAILED_EVENT_NAME, original_username,
-        'Invalid user "' + original_username + '"');
-      return;
-    }
-
     if (!accessGroups || (accessGroups instanceof Array === false) ||
        (accessGroups.length === 0)) {
       throw new Error('Access groups array is not available');
     }
 
     if (!accessGroups.every( (id) => { return id; })) {
-      this.emit(AUTH_FAILED_EVENT_NAME, original_username,
-        'Some access group ids are not defined');
+      this._authErr(new Error('Some access group ids are not defined'));
       return;
     }
 
-    let agroupIds = accessGroups.slice(); // Copy the array
+    let agroupIds = accessGroups.slice();
     agroupIds.sort();
     agroupIds = agroupIds.filter( function(item, index, thisArray) {
       return (index === 0) ? 1 : ((item === thisArray[index - 1]) ? 0 : 1);
     });
     LOGGER.debug('ACCESS GROUP IDS: ' + agroupIds.join(' '));
-    let dbquery = {
-      members:                 username,
-      access_control_group_id: {$in: agroupIds}
-    };
 
-    var countCallback = (err, count) => {
-      if (err) {
-        this.emit(AUTH_FAILED_EVENT_NAME, original_username,
-          'Failed to get authorisation info, DB error: ' + err);
-      } else {
-        if (count == agroupIds.length) {
-          this.emit(AUTH_EVENT_NAME, original_username);
-        } else {
-          this.emit(AUTH_FAILED_EVENT_NAME, original_username,
-            'Not authorised for ' + (count ? 'some' : 'any') + ' of the files');
+    let authurl = config.provide().get('authurl');
+    let reqbody = {groups: agroupIds};
+    let reqopts = url.parse(authurl);
+    reqopts.method = 'POST';
+    if (this.authtype === constants.AUTH_TYPE_TOKEN) {
+      reqbody.token = identifier.trim();
+      reqopts.path = constants.AUTH_URL_TOKEN;
+    } else if (this.authtype === constants.AUTH_TYPE_USER) {
+      reqbody.user = this._getName(identifier);
+      reqopts.path = constants.AUTH_URL_USER;
+    }
+    if (!(reqbody.token || reqbody.user)) {
+      return this._authErr(new Error(`Invalid identifier "${identifier}"`));
+    }
+    let req = http.request(reqopts, ( response ) => {
+      let body = '';
+      response.on('error', this._authErr);
+      response.on('data', (chunk) => {
+        body += chunk;
+      });
+      response.on('end', () => {
+        if (response.statusCode !== 200) {
```

Comment:
As I understand, the error event is triggered by an error in the underlying connection, e.g. sudden disconnect or timeout, not by getting a 404/500/etc.